drivers/dht/dht.c

```diff
@@ -68,7 +68,11 @@ static inline void _reset(dht_t *dev)
 static inline int _wait_for_level(gpio_t pin, bool expect, unsigned timeout)
 {
     while (((gpio_read(pin) > 0) != expect) && timeout) {
-        xtimer_usleep(1);
+/* On 8bit cpu this loop already take more than 1 µs
+ * (3µs on a 16 MHz atmega328p) */
+#if (ARCHITECTURE_WORD_BITS != 8)
```

Comment:
```suggestion
/* On AVR MCUs this loop already take more than 1 µs
 * (3µs on a 16 MHz atmega328p), so we don't need to
 * add further delay */
#ifndef __AVR__
```

This is a nit pick. But it could be that other 8-bit MCUs have a higher clock frequency, a higher IPC, a lower overhead for `gpio_read()`, or whatever. IMO it is better to keep the exception limited to the MCU family we are certain it is needed for.

(Yes, currently AVR is only 8-bit MCU supported. But who knows: Z80 is still pretty popular in the retro community, and there are even new 8051 MCUs with an integrated IEEE 802.15.4 transceiver entering the market...)